src/sparse/KokkosSparse_BsrMatrix.hpp

```diff
@@ -913,6 +941,36 @@ class BsrMatrix {
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
 
+template <typename ScalarType, typename OrdinalType, class Device,
+          class MemoryTraits, typename SizeType>
+void BsrMatrix<ScalarType, OrdinalType, Device, MemoryTraits,
+               SizeType>::ctor_impl(const std::string& /*label*/,
+                                    OrdinalType nrows,
+                                    OrdinalType ncols,
+                                    size_type annz, ScalarType* val,
+                                    OrdinalType* rows, OrdinalType* cols,
+                                    OrdinalType blockDimIn) {
+  numCols_  = ncols;
+  blockDim_ = blockDimIn;
+
+  // Wrap the raw pointers in unmanaged host Views
+  typename values_type::HostMirror unman_val(val, annz);
+  typename row_map_type::HostMirror unman_rows(rows, nrows + 1);
+  typename index_type::HostMirror unman_cols(cols, ncols);
+
+  // Create temporary Views for row_map and entries because the StaticCrsGraph
+  // ctor requires View inputs
+  values_type tmp_row_map("tmp_row_map", nrows + 1);
+  values_type tmp_entries("tmp_entries", ncols);
```

Comment:
Should this be `annz` instead of `ncols`?

Reply:
See updated constructor (as above).